Paco-Server/ear/default/web/js/viz/vizService.js

```diff
@@ -9,91 +10,165 @@ pacoApp.factory('experimentsVizService', ['$http', 'experimentService', function
+      var message = '{ "select":["who"], "query" : { "criteria" : "experiment_id = ?", "values" : [' + experimentId + ']},"group":"who"}';
+    }
+    var participants = httpPostBody(message);
+    return participants;
   }
 
-  function getEvents(id) {
+  function getDateRange(experimentId) {
 
-    var getEvents = $http({
+    var dateRange = [];
+    var format = 'MM/dd/yyyy';
+    if (experimentId != undefined) {
+      var endDateQuery = '{ "select":["response_time"], "query" : { "criteria" : "experiment_id = ?", "values" : [' + experimentId + ']},"order":"response_time desc","limit":"1"}';
+      var startDateQuery = '{ "select":["response_time"], "query" : { "criteria" : "experiment_id = ?", "values" : [' + experimentId + ']},"order":"response_time asc","limit":"1"}';
+    }
+    var startDate = httpPostBody(startDateQuery);
+    var endDate = httpPostBody(endDateQuery);
+
+    startDate.then(function (data) {
+
+      var format_startDate = $filter('date')(new Date(data.data.customResponse[0].response_time), format);
+      dateRange.push(format_startDate)
+    });
+    endDate.then(function (data) {
+      var format_endDate = $filter('date')(new Date(data.data.customResponse[0].response_time), format);
+      dateRange.push(format_endDate)
+    });
+    return dateRange;
+  }
+
+  function getEventsCounts(id) {
+    var eventsCount = [];
+    var getCount = $http({
       method: 'GET',
-      url: '/events?q=experimentId=' + id + '&json'
+      url: '/participantStats?experimentId=' + id + '&reportType=totalEventCounts&statv2=1'
     }).then(function successCallback(response) {
-      return response.data.events;
+      return response;
     }, function errorCallback(error) {
       return error;
     });
+    getCount.then(function (data) {
+      eventsCount.push(data.data[0].schedR, data.data[0].missedR, data.data[0].selfR);
+    });
+    return eventsCount;
+  }
+
+  function getEvents(experimentId, group, participants, startDate, startTime, endDate, endTime) {
+
+    var message = "";
+    if (experimentId != undefined && group != undefined) {
+      message = '{ "query" : { "criteria" : "experiment_id = ? and group_name = ?", "values" : [' + experimentId + ', "' + group + '"]},"limit":"50"}';
+    }
+    if (experimentId != undefined && participants != undefined && participants.length > 0) {
+      var questionMarks = [];
+      var participantsList = [];
+      for (var i = 0; i < participants.length; i++) {
+        questionMarks.push("?");
+        participantsList.push('"' + participants[i] + '"');
+      }
+      message = '{ "query" : { "criteria" : "experiment_id = ? and group_name = ? and who in (' + questionMarks + ')", "values" : [' + experimentId + ', "' + group + '" , ' + participantsList + ']},"limit":"50"}';
+    }
+    if (experimentId != undefined && group != undefined && startDate != undefined) {
+      var start_date = startDate + " " + "00:00:00";
+      var end_date = startDate + " " + "23:59:59";
```

Comment:
Also, in the previous pull request, I asked if having startedDate was correct but now I see that it is trying to span one day in the case that there is a startDate and no endDate.